tofu/spectro/_rockingcurve.py

```diff
@@ -0,0 +1,345 @@
+    # -------------------------------------------------------------------------
+
+    # incident wave polarization (normal & parallel components)
+    polar = np.r_[1., abs(np.cos(2.*theta))]
+    # variables of simplification y, dy, g, L
+    g = psi0_im/(polar*psi_re)
+    y = np.linspace(-10., 10., 501)    # ay
+    dy = np.zeros(501) + 0.1    # day
+    al = np.full((2, 501), 0.)
+
+    power_ratio = np.full((al.shape), np.nan)    # phpo
+    th = np.full((al.shape), np.nan)    # phpo
+    rr = np.full((polar.shape), np.nan)
+    for i in range(al[:, 0].size):
+        al[i, ...] = (y**2 + g[i]**2 + np.sqrt(
+            (y**2 - g[i]**2 + kk**2 - 1.)**2 + 4.*(g[i]*y - rek)**2
+            ))/np.sqrt((kk**2 - 1.)**2 + 4.*rek**2)
+        # reflecting power P_DIN
+        power_ratio[i, ...] = (Fmod/Fbmod)*(
+            al[i, :] - np.sqrt((al[i, :]**2 - 1.))
+        )
+        power_ratiob = power_ratio[i, ...]
+        # intensity scale on the glancing angle (y=kk/g)
+        th[i, ...] = (y*polar[i]*psi_re - psi0_dre)/np.sin(2.*theta)
+        # integration of the power ratio over dy
+        rhy = np.sum(dy*power_ratiob)
+        # diffraction radiation
+        # r(i=0): normal component & r(i=1): parallel component
+        rr[i, ...] = (polar[i]*psi_re*rhy)/np.sin(2.*theta)
+    R_dyn = np.sum(rr)/2.
+    if R_dyn < 1e-7:
+        msg = (
+            "Please check the equations for integrated reflectivity:\n"
+            "the value of R_din ({}) is less than 1e-7.\n".format(R_dyn)
+        )
+        raise Exception(msg)
+
+    fmax = np.max(power_ratio[0] + power_ratio[1])
+    det = (2.*R_dyn)/fmax
+
+    # Plot power ratio
+    # ----------------
+
+    fig1 = plt.figure(figsize=(8, 6))
+    gs = gridspec.GridSpec(1, 1)
+    ax = fig1.add_subplot(gs[0, 0])
+    ax.set_title(
+        'Qz, ' + f'({ih},{ik},{il})' + fr', $\lambda$={lamb} A' +
+        fr', Bragg angle={np.round(theta, decimals=3)} rad'
+    )
+    ax.set_xlabel(r'$\theta$-$\theta_{B}$ (rad)')
+    ax.set_ylabel('P$_H$/P$_0$')
+    ax.plot(th[0, :], power_ratio[0, :], label='normal component')
+    ax.plot(th[1, :], power_ratio[1, :], label='parallel component')
+    ax.legend()
+
+    return (
+        'Wavelength (A):', lamb,
+        'Miller indices:', (ih, ik, il),
+        'Inter-reticular distance (A):', d_atom,
+        'Volume of the unit cell (A^3):', str(np.round(V, decimals=3)),
+        'Bragg angle of reference (rad):', str(np.round(theta, decimals=3)),
+        'Integrated reflectivity, perfect model:',
+        str(np.round(R_per, decimals=8)),
+        'Integrated reflectivity, mosaic model',
+        str(np.round(R_mos, decimals=8)),
+        'Integrated reflectivity, thick crystal model',
+        str(np.round(R_dyn, decimals=8)),
+        'Ratio imag. & real part of structure factor',
+        str(np.round(kk, decimals=3)),
+        'R_perp/R_par', str(np.round(rr[1]/rr[0], decimals=3)),
+        'RC width', str(np.round(det, decimals=6)),
+    )
```

Comment:
Good and detailed,

But :
* replace this long tuple with a dict `dout`
* then add argument `verb=None` (default to True in `_check_inputs()`) for optional pretty-printing of the content of `dout`
* then add argument `returnas=None` (default to `dict` in `_check_inputs()`) to allow for optional returning of `dout`

```
    dout = {
        'Wavelength (A)': lamb,
        'Miller indices': (ih, ik, il),
        'Inter-reticular distance (A)': d_atom,
        'Volume of the unit cell (A^3)': V,
        'Bragg angle of reference (rad)': theta,
        'Integrated reflectivity, perfect model': R_per,
        'Integrated reflectivity, mosaic model': R_mos,
        'Integrated reflectivity, thick crystal model': R_dyn,
        'Ratio imag. & real part of structure factor': kk,
        'R_perp/R_par': rr[1]/rr[0],
        'RC width': det,
    }

    if verb is True:
        lstr = [f'\t- {k0}: {v0}' for k0, v0 in dout.items()]
        msg = (
            "The following data was calculated:\n"
            + "\n".join(lstr)
        )
        print(msg)

    if returnas is dict:
        return dout
```